services/src/main/java/org/keycloak/protocol/oid4vc/tokenstatus/ReferencedTokenValidator.java

```diff
@@ -0,0 +1,353 @@
+    }
+
+    /**
+     * Decodes and decompresses the status list data.
+     *
+     * @param lst The base64url-encoded, DEFLATE-compressed status list
+     * @return The decompressed byte array
+     * @throws ReferencedTokenValidationException if decoding/decompression fails
+     */
+    public static byte[] decodeAndDecompress(String lst) throws ReferencedTokenValidationException {
+        try {
+            // Add padding if necessary for base64url decoding
+            String paddedLst = lst;
+            while (paddedLst.length() % 4 != 0) {
+                paddedLst += "=";
+            }
+
+            // Replace URL-safe characters with standard base64 characters
+            paddedLst = paddedLst.replace('-', '+').replace('_', '/');
+
+            // Decode base64
+            byte[] compressed = Base64.getDecoder().decode(paddedLst);
+
+            // Decompress using DEFLATE with ZLIB wrapper
+            Inflater inflater = new Inflater();
+            inflater.setInput(compressed);
+
+            byte[] decompressed = new byte[compressed.length * 10]; // Initial buffer size
+            int decompressedLength = inflater.inflate(decompressed);
+            inflater.end();
+
+            // Create a new array with the exact size
+            byte[] result = new byte[decompressedLength];
+            System.arraycopy(decompressed, 0, result, 0, decompressedLength);
+
+            return result;
+
+        } catch (IllegalArgumentException e) {
+            throw new ReferencedTokenValidationException("Failed to decode base64url status list", e);
+        } catch (DataFormatException e) {
+            throw new ReferencedTokenValidationException("Failed to decompress status list", e);
+        }
+    }
+
+    /**
+     * Reads a status value from the status list at the specified index.
+     *
+     * @param lst  The base64url-encoded, DEFLATE-compressed status list
+     * @param idx  The token index to read
+     * @param bits The number of bits per status (1, 2, 4, or 8)
+     * @return The status value at the given index
+     * @throws ReferencedTokenValidationException if the operation fails
+     */
+    private int readStatusValue(String lst, int idx, int bits) throws ReferencedTokenValidationException {
+        byte[] bytes = decodeAndDecompress(lst);
+        return readStatusValueFromBytes(bytes, idx, bits);
+    }
+
+    /**
+     * Helper method to read status value from a byte array.
+     *
+     * @param bytes The decompressed byte array
+     * @param idx   The token index to read
+     * @param bits  The number of bits per status (1, 2, 4, or 8)
+     * @return The status value at the given index
+     * @throws ReferencedTokenValidationException if the operation fails
+     */
+    public static int readStatusValueFromBytes(byte[] bytes, int idx, int bits) throws ReferencedTokenValidationException {
+        if (bits != 1 && bits != 2 && bits != 4 && bits != 8) {
```

Comment:
We commonly use this alternative for concision.
```suggestion
        if (!List.of(1, 2, 4, 8).contains(bits)) {
```